spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/SpannerPrimaryKeyColumn.java

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core.mapping;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation for a {@link SpannerPersistentEntity} that allows specifying the primary key
+ * columns and their order.
+ *
+ * @author Chengyuan Zhao
+ */
+@Documented
+@Target(ElementType.FIELD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface SpannerPrimaryKeyColumn {
```

Comment:
You could consider adding a `value()` alias so the annotation can be expressed as `@PrimaryKeyColumn(1)`.

Example from Cassandra:

```java
	/**
	 * The name of the column in the table.
	 */
	@AliasFor(attribute = "name")
	String value() default "";

	/**
	 * The name of the column in the table.
	 */
	@AliasFor(attribute = "value")
	String name() default "";
```

Reply:
Good idea! Will do.